source/common/stats/tag_extractor_impl.h

```diff
@@ -14,6 +18,35 @@
 namespace Envoy {
 namespace Stats {
 
+// To check if a tag extractor is actually used you can run
+// bazel test //test/... --test_output=streamed --define=perf_annotation=enabled
+#ifdef ENVOY_PERF_ANNOTATION
+
+struct Counters {
+  uint32_t skipped_{};
+  uint32_t matched_{};
+  uint32_t missed_{};
+};
+
+#define PERF_TAG_COUNTERS                                                                          \
+  ~TagExtractorImplBase() override {                                                               \
```

Comment:
Nit: putting the whole destructor here is a bit magical, I think I would prefer it under an explicit `#ifdef`

Reply:
done